CMakeLists.txt

```diff
@@ -0,0 +1,200 @@
+else()
+  add_definitions(-Wall)
+  add_definitions(-std=c++11)
+  if (TEST_BUILD)
+    if (NOT APPLE)
+      add_definitions(-Werror)
+    endif()
+    add_definitions(-pedantic)
+  endif()
+endif()
+
+option(EXTERNAL_LIBOSMIUM "Do not use the bundled libosmium" OFF)
+
+#############################################################
+# Detect available headers and set global compiler options
+#############################################################
+
+INCLUDE (CheckIncludeFiles)
+INCLUDE (CheckFunctionExists)
+INCLUDE (CheckTypeSize)
+
+ADD_DEFINITIONS( -DOSM2PGSQL_DATADIR=${DATA_DIR} )
+ADD_DEFINITIONS( -DFIXED_POINT )
+
+CHECK_INCLUDE_FILES (termios.h HAVE_TERMIOS_H)
+CHECK_INCLUDE_FILES (libgen.h HAVE_LIBGEN_H)
+CHECK_INCLUDE_FILES (unistd.h HAVE_UNISTD_H)
+CHECK_INCLUDE_FILES (sys/wait.h HAVE_SYS_WAIT_H)
+CHECK_INCLUDE_FILES (sys/time.h HAVE_SYS_TIME_H)
+CHECK_INCLUDE_FILES (sys/mman.h HAVE_MMAP)
+
+if (WIN32)
+  set(HAVE_LIBGEN_H FALSE)
+endif()
+
+CHECK_FUNCTION_EXISTS(lseek64 HAVE_LSEEK64)
+CHECK_FUNCTION_EXISTS(posix_fallocate HAVE_POSIX_FALLOCATE)
+CHECK_FUNCTION_EXISTS(posix_fadvise HAVE_POSIX_FADVISE)
+CHECK_FUNCTION_EXISTS(sync_file_range HAVE_SYNC_FILE_RANGE)
+CHECK_FUNCTION_EXISTS(fork HAVE_FORK)
+
+CHECK_TYPE_SIZE("off_t" SIZEOF_OFF_T)
+
+if (NOT WIN32 AND NOT APPLE AND NOT HAVE_LSEEK64 AND NOT SIZEOF_OFF_T EQUAL 8)
+  message(FATAL_ERROR "Flat nodes cache requires a 64 bit capable seek")
+endif()
+
+#############################################################
+# Find necessary libraries
+#############################################################
+
+if (NOT EXTERNAL_LIBOSMIUM)
+  set(OSMIUM_INCLUDE_DIR "contrib/libosmium")
+endif()
+
+find_package(Osmium REQUIRED COMPONENTS io geos proj)
+include_directories(SYSTEM ${OSMIUM_INCLUDE_DIRS})
+
+find_package(Lua52)
+if (LUA_LIBRARIES)
+include_directories(${LUA_INCLUDE_DIR})
+set(HAVE_LUA 1)
+else()
+  message(STATUS "lua libraries not found. You will NOT be able to use lua scripts for tag transform.")
+endif()
+
+if (MSVC)
+  # Boost thread needs extra libraries
+  set(BOOST_EXTRA date_time chrono)
+endif()
+
+# first try to find the version
+find_package(Boost 1.50 REQUIRED COMPONENTS system filesystem thread ${BOOST_EXTRA})
+include_directories(${Boost_INCLUDE_DIR})
+
+find_package(PostgreSQL REQUIRED)
+include_directories(${PostgreSQL_INCLUDE_DIRS})
+
+if (NOT PostgreSQL_FOUND)
+  message(FATAL_ERROR "Can not find PostgreSQL. Do not forget to install all postgresql-dev packages, including server-dev. Try using options like -DPostgreSQL_TYPE_INCLUDE_DIR=/usr/include/postgresql/9.4/server/catalog on Ubuntu")
+endif()
+
+find_package(Threads)
+
+if (NOT CMAKE_COMPILER_IS_GNUCC AND WIN32)
+  find_path(GETOPT_INCLUDE_DIR getopt.h)
+  find_library(GETOPT_LIBRARY NAMES wingetopt )
+  if (GETOPT_INCLUDE_DIR AND GETOPT_LIBRARY)
+    include_directories(${GETOPT_INCLUDE_DIR})
+  endif()
+endif()
+
+if (WIN32)
```

Comment:
And I'll remove ``if (NOT PostgreSQL_FOUND)`` inserting help for getting wingetopt.